cluster/engine.go

```diff
@@ -417,10 +420,20 @@ func (e *Engine) CheckConnectionErr(err error) {
 	if err == nil {
 		e.setErrMsg("")
 		// If current state is unhealthy, change it to healthy
+		// A lock is needed to avoid race condition, which can lead to block on resumeCh
+		// "changed" flag will take care of slow operations, ex: emitEvent
```

Comment:
I don't really understand this comment. `changed` doesn't affect how we call `emitEvent` at all.

Reply:
`changed` which is set to true inside the `if e.state == stateUnhealthy` block and in the thread safe block, will be used to decide whether to run (outside the lock) slow operations : ex:  emitEvent (which ends up calling the engine)

Reply:
Can we update this comment to be more clear, then?

Reply:
Sure, I guess It can be a bit clearer